src/test/java/com/fasterxml/jackson/databind/ObjectReaderTest.java

```diff
@@ -805,4 +808,84 @@ public int hashCode() {
             return _delegate.hashCode();
         }
     }
+
+
+    public void testReadValueFromFile() throws Exception {
+        File file = _createFileWithNameAndJson(
+            "testReadValueFromFile.json",
+            a2q("{ 'name': 'John Doe'}"));
+
+        FilePerson bean = newJsonMapper().readerFor(FilePerson.class).readValue(file);
+
+        assertEquals("John Doe", bean.name);
+        _cleanUpFromFileSystem(file);
+    }
+
+    public void testReadValueFromFile2() throws Exception {
+        File file = _createFileWithNameAndJson(
+            "testReadValueFromFile2.json",
+            a2q("{ 'name': 'John Doe'}"));
+
+        FilePerson bean = newJsonMapper().reader().readValue(file, FilePerson.class);
+
+        assertEquals("John Doe", bean.name);
+        _cleanUpFromFileSystem(file);
+    }
+
+    public void testReadValueFromNonExistentFile() throws Exception {
+        File file = new File("testReadValueFromNonExistentFile.json");
+        try {
+            newJsonMapper().readValue(file, FilePerson.class);
+            fail("should not pass");
+        } catch (FileNotFoundException e) {
+            verifyException(e, "testReadValueFromNonExistentFile");
+        }
+    }
+
+    public void testInputStreamFromEmptyFile() throws Exception {
+        File file = new File("testInputStreamFromEmptyFile.json");
+        assertTrue(file.createNewFile());
+
+        try {
+            newJsonMapper().readerFor(FilePerson.class).readValue(file);
+            fail("should not pass");
+        } catch (MismatchedInputException e) {
+            verifyException(e, "No content to map due to end-of-input");
+        } finally {
+            _cleanUpFromFileSystem(file);
+        }
+    }
+
+    public void testReadValuesFromFile() throws Exception {
+        File file = _createFileWithNameAndJson(
+            "testReadValuesFromFile.json",
+            a2q("{ 'name': 'One'} { 'name': 'Two'}"));
+
+        MappingIterator<FilePerson> iterator = newJsonMapper().readerFor(FilePerson.class).readValues(file);
+
+        _verifyWithMappingIterator(iterator, "One", "Two");
+        _cleanUpFromFileSystem(file);
+    }
+
+    private void _verifyWithMappingIterator(MappingIterator<FilePerson> iterator, String... names) throws Exception {
+        for (String n : names) {
+            assertEquals(n, iterator.next().name);
+        }
+        assertFalse(iterator.hasNext());
+        iterator.close();
+    }
+
+    private File _createFileWithNameAndJson(String fileName, String json) throws Exception {
```

Comment:
So, 2 improvements:

1. Use "File.createTempFile()" (like https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createTempFile-java.lang.String-java.lang.String-) to avoid creating these on current directory
2. Mark them for deletion using "File.deleteOnExit()" (https://docs.oracle.com/javase/8/docs/api/java/io/File.html#deleteOnExit--) just in case test crashes or whatever

(there's an example in `jackson-core` under `src/test/java/com/fasterxml/jackson/core/json/JsonFactoryTest.java` of (1) but it's quite straight-forward)

Reply:
@JooHyukKim Great, happy to be of some use here :)

And once again, thank you for submitting PRs to improve things like testing.